src/pykiso/lib/connectors/cc_serial.py

```diff
@@ -125,25 +129,31 @@ def __init__(
         )
 
         self.current_write_timeout = write_timeout
-        self.serial.port = self._get_port(port=port, vid=vid, pid=pid)
+        self.serial.port = self._get_port(
+            port=port, vid=vid, pid=pid, serial_number=serial_number
+        )
 
     @staticmethod
-    def _find_device(vid: int, pid: int) -> str:
-        """Return the device which matches pid and vid.
+    def _find_device(vid: int, pid: int, serial_number: str) -> str:
+        """Return the device which matches the given pid/vid or serial_number.
 
         :param vid: vendor id
         :param pid: product id
+        :param serial_number: serial number
 
         :return: com port for the found device. I.e. "COM4" or "/dev/tty1"
         """
 
         attached_com_devices = serial.tools.list_ports.comports()
         found_devices = [
-            port for port in attached_com_devices if port.pid == pid and port.vid == vid
+            port
+            for port in attached_com_devices
+            if (port.pid == pid and port.vid == vid and pid is not None)
```

Comment:
This condition is hard to read, I pretty sure you you can split it. I really love the list comprehension but adding too much condition decrease the readability.
You test the pid is not None but what bout the vid?